src/main/java/com/textmessenger/service/MessageServiceImpl.java

```diff
@@ -38,5 +42,10 @@ public void updateMessage(long id, Message message) {
   public void deleteMessage(long id) {
     messageRepository.delete(messageRepository.getOne(id));
   }
+
+  @Override
+  public List<Message> getMessagesFromDialog(Dialog dialog) {
+    return messageRepository.findAll().stream().filter(e -> e.getDialog().equals(dialog)).collect(Collectors.toList());
```

Comment:
What if you have one billion messages? This should be done on a DB level using Spring Data interface